src/Main.elm

```diff
@@ -1,16 +1,102 @@
 module Main exposing (main)
 
-import Html exposing (Html)
-
-
 -- Import Modules
 
-import Data.Audience
-import Data.AudienceFolder
+import Data.Audience exposing (audiences)
+import Data.AudienceFolder exposing (audienceFolders)
+import Html exposing (Attribute, Html, text)
+import Html.Attributes exposing (style)
+import Html.Events exposing (onClick)
+import Lazy.Tree.Zipper as Zipper exposing (Zipper)
+import NavTree exposing (TreeItem, isFolderWithId)
 
 
 {-| Main file of application
 -}
-main : Html msg
+main : Program Never Model Msg
 main =
-    Html.text "There will be app soon!"
+    Html.beginnerProgram
+        { model = model
+        , view = view
+        , update = update
+        }
+
+
+model : Model
+model =
+    NavTree.build audienceFolders audiences
+        |> Zipper.fromTree
+
+
+type Msg
+    = GoUp
+    | OpenFolder Int
+
+
+type alias Model =
+    Zipper TreeItem
+
+
+update : Msg -> Model -> Model
+update msg zipper =
+    case msg of
+        GoUp ->
+            Zipper.up zipper
+                -- Nothing means we're already at the root, OK to stay there
+                |> Maybe.withDefault zipper
```

Comment:
This is fine in this case anyway Zipper has helper function `attempt` so less verbose equivalent would be `Zipper.attempt Zipper.up zipper`

Reply:
Good tip, didn't notice that exists. I like reading the docs before I start using new package, but in this case I failed to finish reading everything :)

Reply:
yeah that surface area of that package is quite big :/